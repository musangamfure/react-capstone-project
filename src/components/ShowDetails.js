import React from 'react';
import { Link, useParams } from 'react-router-dom';
import { useSelector } from 'react-redux';
import '../styles/ShowDetail.css';

function ShowDetails() {
  const { id } = useParams();
  const shows = useSelector((state) => state.shows.Shows);

  const show = shows.find((show) => show.id.toString() === id);

  if (!show) {
    return <h2>Loading Data</h2>;
  }

  const {
    name, rating, category, summary, image, official, language,
  } = show;
  const Summary = summary
    .replace(/<p>/g, '')
    .replace(/<\/p>/g, '')
    .replace(/<b>/g, '')
    .replace(/<\/b>/g, '');

  return (
    <div className="show-details">
      <Link className="back-button" to="/">
        <i className="fa-solid fa-arrow-left" />
        {' '}
        Back
      </Link>
      <div className="details-container">
        <div className="headings">
          <h2>{name}</h2>
          <div className="rating">
            <h3>
              Rating:
              {rating.average}
            </h3>
          </div>
        </div>
        <div className="show-main">
          <img src={image.medium} alt="" />
          <div className="right-section">
            <p>{Summary}</p>
            <div className="show-categories">
              {category.map((categorys) => (
                <span key={categorys}>
                  <b>{categorys}</b>
                </span>
              ))}
            </div>
            <p>
              Language:
              {language}
            </p>
            <div className="official-website">
              <a href={official} target="_blank" rel="noreferrer">
                <p>Visit Official Website </p>
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  );
}

export default ShowDetails;
