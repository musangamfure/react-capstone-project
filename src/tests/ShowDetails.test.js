import React from 'react';
import { render } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import ShowDetails from '../components/ShowDetails';

// Create a mock Redux store
const mockStore = configureStore([]);
const store = mockStore({
  shows: {
    Shows: [
      {
        id: '1',
        name: 'Show Name',
        rating: {
          average: 8.5,
        },
        category: ['Drama', 'Thriller'],
        summary: 'Summary of the show',
        image: {
          medium: 'abc.jpg',
        },
        official: 'https://official-website.com',
        language: 'English',
      },
    ],
  },
});

test('renders ShowDetail component', () => {
  const { container } = render(
    <Provider store={store}>
      <Router>
        <ShowDetails />
      </Router>
    </Provider>,
  );

  expect(container.innerHTML).toMatchSnapshot();
});
