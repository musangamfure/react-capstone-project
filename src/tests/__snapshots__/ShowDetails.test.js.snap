// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders ShowDetail component 1`] = `"<h2>Loading Data</h2>"`;
